### PR TITLE
ENTESB-10252: fixed [FIS]no_proxy in jvm argument not honoured in FIS…

### PR DESCRIPTION
**Pull request by valdar**
… Image.

### CODE REVIEW

File: fish-pepper/run-java-sh/fp-files/run-java.sh [hunk 1]
```diff
@@ -601,7 +601,7 @@ run() {
   # Don't put ${args} in quotes, otherwise it would be interpreted as a single arg.
   # However it could be two args (see above). zsh doesn't like this btw, but zsh is not
   # supported anyway.
-  echo exec $(exec_args) java $(java_options) -cp "$(classpath)" ${args} "$@"
+  echo exec $(exec_args) java "$(java_options)" -cp "$(classpath)" ${args} "$@"
```
Comment (rhuss) [inline fish-pepper/run-java-sh/fp-files/run-java.sh]: You don't need the quotes here and it doesn't reflect the code which is executed.

File: fish-pepper/run-java-sh/fp-files/run-java.sh [hunk 2]
```diff
@@ -483,7 +483,7 @@ proxy_options() {
 
   local noProxy="${no_proxy:-${NO_PROXY:-}}"
   if [ -n "$noProxy" ] ; then
-    ret="$ret -Dhttp.nonProxyHosts=\"$(echo "|$noProxy" | sed -e 's/,[[:space:]]*/|/g' | sed -e 's/|\./|\*\./g' | cut -c 2-)\""
+    ret="$ret -Dhttp.nonProxyHosts=$(echo "|$noProxy" | sed -e 's/,[[:space:]]*/|/g' | sed -e 's/|\./|\*\./g' | cut -c 2-)"
```
Comment (rhuss) [inline fish-pepper/run-java-sh/fp-files/run-java.sh]: Not sure whether there can be still spaces in a noProxy spec, so still keeping the quotes would be safe in any case. Let's find out why the test is failing when quotes are provided.

File: test/t/04_proxy.bats [hunk 3]
```diff
@@ -54,7 +54,7 @@ load test_helper
   echo $status
   echo $output
 
-  assert_jvmarg "-Dhttp.nonProxyHosts=\"localhost\|host.example.com\""
+  assert_jvmarg "-Dhttp.nonProxyHosts=localhost\|host.example.com"
```
Comment (rhuss) [inline test/t/04_proxy.bats]: See above. We probably have to adopt the `assert_jvmarg` function.

File: test/t/04_proxy.bats [hunk 4]
```diff
@@ -69,25 +69,21 @@ load test_helper
   echo $status
   echo $output
 
-  assert_jvmarg "-Dhttp.nonProxyHosts=\"localhost\|\*.example.com\""
+  assert_jvmarg "-Dhttp.nonProxyHosts=localhost\|\*.example.com"
```
Comment (rhuss) [inline test/t/04_proxy.bats]: See above

File: test/t/04_proxy.bats [hunk 5]
```diff
@@ -69,25 +69,21 @@ load test_helper
...
   d=$(mktmpdir "proxy")
 
-  cp "$TEST_JAR_DIR/test.jar" "$d/test.jar"
-  no_proxy='.example.com, localhost,.cluster.local' JAVA_APP_DIR=$d run $TEST_SHELL $RUN_JAVA
+  cp "$TEST_JAR_DIR/TestProxy.class" "$d/TestProxy.class"
+  HTTP_PROXY=http://dummy.com HTTPS_PROXY=http://dummy.com NO_PROXY=.google.com\|.redhat.com no_proxy=.google.com\|.redhat.com JAVA_MAIN_CLASS=TestProxy JAVA_APP_DIR=$d run $TEST_SHELL $RUN_JAVA
   echo $status
   echo $output
 
-  assert_jvmarg "-Dhttp.nonProxyHosts=\"\*.example.com\|localhost\|\*.cluster.local\""
```
Comment (rhuss) [inline test/t/04_proxy.bats]: No assertion on the test output ?

File: test/t/04_proxy.bats [hunk 6]
```diff
@@ -100,7 +96,7 @@ load test_helper
 
   assert_jvmarg "-Dhttp.proxyHost=proxy -Dhttp.proxyPort=3128"
   assert_jvmarg "-Dhttps.proxyHost=proxy -Dhttps.proxyPort=4128"
-  assert_jvmarg "-Dhttp.nonProxyHosts=\"localhost\|host.example.com\""
+  assert_jvmarg "-Dhttp.nonProxyHosts=localhost\|host.example.com"
```
Comment (rhuss) [inline test/t/04_proxy.bats]: see above

File: fish-pepper/run-java-sh/fp-files/run-java.sh (hunk 2 above)
Comment (rhuss) [reply fish-pepper/run-java-sh/fp-files/run-java.sh]: Not sure what the actual fix is. Is it just removing the quotes ?

File: fish-pepper/run-java-sh/fp-files/run-java.sh (hunk 1 above)
Comment (valdar) [reply fish-pepper/run-java-sh/fp-files/run-java.sh]: agreed and removed

File: fish-pepper/run-java-sh/fp-files/run-java.sh (hunk 2 above)
Comment (valdar) [reply fish-pepper/run-java-sh/fp-files/run-java.sh]: yes this is the actual fix

**Pull request merged**